src/EnergyPlus/SystemAvailabilityManager.cc

```diff
@@ -358,73 +350,58 @@ namespace SystemAvailabilityManager {
-        cCurrentModuleObject = "AvailabilityManager:NightCycle";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, cCurrentModuleObject, numArgs, NumAlphas, NumNumbers);
-        maxNumbers = max(maxNumbers, NumNumbers);
-        maxAlphas = max(maxAlphas, NumAlphas);
-        cCurrentModuleObject = "AvailabilityManager:DifferentialThermostat";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, cCurrentModuleObject, numArgs, NumAlphas, NumNumbers);
-        maxNumbers = max(maxNumbers, NumNumbers);
-        maxAlphas = max(maxAlphas, NumAlphas);
-        cCurrentModuleObject = "AvailabilityManager:HighTemperatureTurnOff";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, cCurrentModuleObject, numArgs, NumAlphas, NumNumbers);
-        maxNumbers = max(maxNumbers, NumNumbers);
-        maxAlphas = max(maxAlphas, NumAlphas);
-        cCurrentModuleObject = "AvailabilityManager:HighTemperatureTurnOn";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, cCurrentModuleObject, numArgs, NumAlphas, NumNumbers);
-        maxNumbers = max(maxNumbers, NumNumbers);
-        maxAlphas = max(maxAlphas, NumAlphas);
-        cCurrentModuleObject = "AvailabilityManager:LowTemperatureTurnOff";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, cCurrentModuleObject, numArgs, NumAlphas, NumNumbers);
-        maxNumbers = max(maxNumbers, NumNumbers);
-        maxAlphas = max(maxAlphas, NumAlphas);
-        cCurrentModuleObject = "AvailabilityManager:LowTemperatureTurnOn";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, cCurrentModuleObject, numArgs, NumAlphas, NumNumbers);
-        maxNumbers = max(maxNumbers, NumNumbers);
-        maxAlphas = max(maxAlphas, NumAlphas);
-        cCurrentModuleObject = "AvailabilityManager:NightVentilation";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, cCurrentModuleObject, numArgs, NumAlphas, NumNumbers);
-        maxNumbers = max(maxNumbers, NumNumbers);
-        maxAlphas = max(maxAlphas, NumAlphas);
-        cCurrentModuleObject = "AvailabilityManager:OptimumStart";
-        state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, cCurrentModuleObject, numArgs, NumAlphas, NumNumbers);
-        maxNumbers = max(maxNumbers, NumNumbers);
-        maxAlphas = max(maxAlphas, NumAlphas);
+        for (int currentModuleObjectCount = 0; currentModuleObjectCount < static_cast<int>(DataPlant::SystemAvailabilityType::Num); ++currentModuleObjectCount) {
+            std::string_view cCurrentModuleObject = SystemAvailabilityTypeNamesCC[currentModuleObjectCount];
+            state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(state, cCurrentModuleObject, numArgs, NumAlphas, NumNumbers);
+            maxNumbers = max(maxNumbers, NumNumbers);
+            maxAlphas = max(maxAlphas, NumAlphas);
+        }
```

Comment:
Iterating through `enum class`, using `constexpr string_view`s instead of `string`s.